packages/dma-contracts/contracts/actions/aave/v3l2/Borrow.sol

```diff
@@ -25,25 +25,32 @@ interface IL2Encoder {
 
 contract AaveV3L2Borrow is Executable, UseStore {
   using Write for OperationStorage;
+  using Read for OperationStorage;
 
   constructor(address _registry) UseStore(_registry) {}
 
   /**
    * @param data Encoded calldata that conforms to the BorrowData struct
    */
-  function execute(bytes calldata data, uint8[] memory) external payable override {
+  function execute(bytes calldata data, uint8[] memory paramsMap) external payable override {
     BorrowData memory borrow = parseInputs(data);
 
+    uint256 mappedBorrowAmount = store().readUint(
+      bytes32(borrow.amount),
+      paramsMap[1],
+      address(this)
+    );
+
     IL2Pool(registry.getRegisteredService(AAVE_POOL)).borrow(
       IL2Encoder(registry.getRegisteredService(AAVE_L2_ENCODER)).encodeBorrowParams(
         borrow.asset,
-        borrow.amount,
+        mappedBorrowAmount,
         2,
         0
       )
     );
 
-    store().write(bytes32(borrow.amount));
+    store().write(bytes32(mappedBorrowAmount));
```

Comment:
same comment, if it can be skipped if it's already stored then it should be  done ( and offchain code adjusted as required - re: indices)